lib/node_modules/@stdlib/math/base/special/ldexpf/src/main.c

```diff
@@ -60,52 +65,47 @@ float stdlib_base_ldexpf( const float frac, const int32_t exp ) {
 	stdlib_base_float32_to_word( frac, &uix );
 	ix = (int32_t)uix;
 
-	// Extract exponent
-	k = ( ix & 0x7f800000 ) >> 23;
+	// Extract exponent:
+	k = ( ix & STDLIB_CONSTANT_FLOAT32_EXPONENT_MASK ) >> ( STDLIB_CONSTANT_FLOAT32_PRECISION - 1 );
 
-	// 0 or subnormal frac
+	// 0 or subnormal frac:
 	fracc = frac;
 	if ( k == 0 ) {
-		if ( ( ix & 0x7fffffff ) == 0 ) {
-			// +-0
+		if ( ( ix & STDLIB_CONSTANT_FLOAT32_ABS_MASK ) == 0 ) {
+			// +-0:
 			return frac;
 		}
 		fracc = frac * TWO25;
 		stdlib_base_float32_to_word( fracc, &uix );
 		ix = (int32_t)uix;
-		k = ( ( ix & 0x7f800000 ) >> 23 ) - 25;
+		k = ( ( ix & STDLIB_CONSTANT_FLOAT32_EXPONENT_MASK ) >> ( STDLIB_CONSTANT_FLOAT32_PRECISION - 1 ) ) - 25;
 		if ( exp < -50000 ) {
-			// Underflow
+			// Underflow:
 			return 0.0;
 		}
 	}
 
-	// NaN or Inf
-	if ( k == 0xff ) {
+	// NaN or Inf:
+	if ( k == ALL_ONES ) {
 		return fracc + fracc;
 	}
 	k += exp;
-	if ( k > 0xfe ) {
-		// Overflow
+	if ( k > ALL_ONES - 1 ) {
+		// Overflow:
 		return stdlib_base_copysignf( STDLIB_CONSTANT_FLOAT32_PINF, fracc );
 	}
 	if ( k > 0 ) {
-		// Normal result
-		stdlib_base_float32_from_word( (uint32_t)( ( ix & 0x807fffff ) | ( k << 23 ) ), &fracc );
+		// Normal result:
+		stdlib_base_float32_from_word( (uint32_t)( ( ix & FLOAT32_SIGNIFICAND_MASK_WITH_SIGN ) | ( k << ( STDLIB_CONSTANT_FLOAT32_PRECISION - 1 ) ) ), &fracc );
 		return fracc;
 	}
 	if ( k <= -25 ) {
-		if ( exp > 50000 ) {
```

Comment:
@gunjjoshi Same comment as for the JS version.

Reply:
Okay. I see what this was in `scalbn`. The issue is when `exp` is so large that `k` integer overflows. Recall that `k` is `int32_t`. When `k` exceeds the maximum positive signed 32-bit integer, the value wraps around. In which case, L92 can generate a negative `k` when `exp` is very large. You need to revert this change.

Reply:
...and add a respective test.